fairscale/nn/misc/checkpoint_activations.py

```diff
@@ -41,8 +40,23 @@ def checkpoint_wrapper(module: nn.Module, offload_to_cpu: bool = False) -> nn.Mo
         (nn.Module):
             wrapped module
     """
-    module.forward = functools.partial(_checkpointed_forward, module.forward, offload_to_cpu)  # type: ignore
-    return module
+    # Do not use functools.partial like:
+    #
+    #     module.forward = functools.partial(_checkpointed_forward, module.forward, offload_to_cpu)
+    #
+    # It causes the backward to hold-on to tensor memory even when model is
+    # freed.
```

Comment:
How did you test this? I'm not able to repro this using this simple test script (i.e., I get the same results regardless of whether it uses functools.partial or the Module version): https://gist.github.com/myleott/7d2ea5a8d27cb6443948e9fa173faf8f